apis/v1alpha1/version.go

```diff
@@ -0,0 +1,197 @@
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VersionSpec defines the desired state of Version.
+//
+
+type VersionSpec struct {
+
+	// Only publish a version if the hash value matches the value that's specified.
+	// Use this option to avoid publishing a version if the function code has changed
+	// since you last updated it. You can get the hash for the version that you
+	// uploaded from the output of UpdateFunctionCode.
+	CodeSHA256 *string `json:"codeSHA256,omitempty"`
+	// A description for the version to override the description in the function
+	// configuration.
+	Description *string `json:"description,omitempty"`
+	// The name of the Lambda function.
+	//
+	// Name formats
+	//
+	//   - Function name - MyFunction.
+	//
+	//   - Function ARN - arn:aws:lambda:us-west-2:123456789012:function:MyFunction.
+	//
+	//   - Partial ARN - 123456789012:function:MyFunction.
+	//
+	// The length constraint applies only to the full ARN. If you specify only the
+	// function name, it is limited to 64 characters in length.
+	// +kubebuilder:validation:Required
+	FunctionName *string `json:"functionName"`
+	// Only update the function if the revision ID matches the ID that's specified.
+	// Use this option to avoid publishing a version if the function configuration
+	// has changed since you last updated it.
+	RevisionID *string `json:"revisionID,omitempty"`
+}
+
+// VersionStatus defines the observed state of Version
+type VersionStatus struct {
+	// All CRs managed by ACK have a common `Status.ACKResourceMetadata` member
+	// that is used to contain resource sync state, account ownership,
+	// constructed ARN for the resource
+	// +kubebuilder:validation:Optional
+	ACKResourceMetadata *ackv1alpha1.ResourceMetadata `json:"ackResourceMetadata"`
+	// All CRS managed by ACK have a common `Status.Conditions` member that
+	// contains a collection of `ackv1alpha1.Condition` objects that describe
+	// the various terminal states of the CR and its backend AWS service API
+	// resource
+	// +kubebuilder:validation:Optional
+	Conditions []*ackv1alpha1.Condition `json:"conditions"`
+	// The instruction set architecture that the function supports. Architecture
+	// is a string array with one of the valid values. The default architecture
+	// value is x86_64.
+	// +kubebuilder:validation:Optional
+	Architectures []*string `json:"architectures,omitempty"`
+	// The size of the function's deployment package, in bytes.
+	// +kubebuilder:validation:Optional
+	CodeSize *int64 `json:"codeSize,omitempty"`
+	// The function's dead letter queue.
+	// +kubebuilder:validation:Optional
+	DeadLetterConfig *DeadLetterConfig `json:"deadLetterConfig,omitempty"`
```

Comment:
Most of the fields in here are not really relevant to a function, should we get rid of them?

Reply:
We're using [`PublishVersion`](https://docs.aws.amazon.com/lambda/latest/dg/API_PublishVersion.html) API to set the status and its response has all these fields. So we cannot do anything with the extra fields.